dashboard/app/models/script.rb

```diff
@@ -436,6 +448,51 @@ def self.get_from_cache(id_or_name, version_year: nil)
     end
   end
 
+  def self.get_family_without_cache(family_name)
+    Script.where(family_name: family_name).order("properties -> '$.version_year' DESC")
+  end
+
+  # Returns all scripts within a family from the Rails cache.
+  # Populates the cache with scripts in that family upon cache miss.
+  # @param family_name [String] Family name for the desired scripts.
+  # @return [Array<Script>] Scripts within the specified family.
+  def self.get_family_from_cache(family_name)
+    return Script.get_family_without_cache(family_name) unless should_cache?
+
+    script_family_cache.fetch(family_name) do
+      # Populate cache on miss.
+      script_cache[family_name] = Script.get_family_without_cache(family_name)
+    end
+  end
+
+  def self.get_script_family_redirect_for_user(family_name, user: nil, locale: 'en-US')
+    return nil unless family_name
+
+    family_scripts = Script.get_family_from_cache(family_name).sort_by(&:version_year).reverse
+
+    if user
+      assigned_script_ids = user.section_scripts.pluck(:id)
+      script_name = family_scripts.select {|s| assigned_script_ids.include?(s.id)}&.first&.name
+      return Script.new(redirect_to: script_name) if script_name
+    end
+
+    locale_str = locale&.to_s
+    latest_version = nil
+    family_scripts.each do |script|
+      next unless script.is_stable
+      latest_version ||= script
+
+      is_supported = script.supported_locales&.include?(locale_str) || locale_str&.downcase&.start_with?('en')
+      if is_supported
+        latest_version = script if is_supported
```

Comment:
❤️ the self-documenting variable name! questions/comments here:
1. remove redundant double-checking of `if is_supported` on this line
2. does the `start_with?('en')` capture just one locale, or several? it makes sense that a user in England would be able to use an American-English version of our course... just requesting that the intention be documented with a quick comment.

Reply:
thank you! all english-speaking locales should be supported, so i'll add a comment around that